src/test/ui/attributes/register-attr-tool.rs

```diff
@@ -0,0 +1,19 @@
+// check-pass
+// compile-flags: --cfg foo
+
+#![feature(register_attr)]
+#![feature(register_tool)]
+
+#![register_attr(attr)]
+#![register_tool(tool)]
+#![register_tool(rustfmt, clippy)] // OK
+#![cfg_attr(foo, register_attr(conditional_attr))]
+#![cfg_attr(foo, register_tool(conditional_tool))]
+
+#[attr]
+#[tool::attr]
```

Comment:
what if there's just `#[tool]` ?

Reply:
```rust
#[rustfmt] // error: cannot find attribute `rustfmt` in this scope
fn main() {}
```
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=a34779363ede9113bf4f8574f69f42da

Reply:
We already have a test case for this in `ui/tool-attributes/tool-attributes-misplaced-1.rs`.